### PR TITLE
first iteration

### PR DESCRIPTION
**Pull request by dranikpg**
- two small issues with hset commands
- we have very little support for server-management commands, most of them are stubs / don't react to help either, I removed the subcommands

### CODE REVIEW

File: docs/command-reference/server-management/client-list.md [hunk 1]
```diff
@@ -34,58 +31,12 @@ Here is the meaning of the fields:
...
-u: the client is unblocked
-U: the client is connected via a Unix domain socket
-x: the client is in a MULTI/EXEC context
-t: the client enabled keys tracking in order to perform client side caching
-R: the client tracking target client is invalid
-B: the client enabled broadcast tracking mode 
-```
-
-The file descriptor events can be:
+* `phase`: TODO
```
Comment (dranikpg) [inline docs/command-reference/server-management/client-list.md:34]: How should I describe this?

"whether the connection is currently processing a command"

The states are: process and readsock.

File: docs/command-reference/server-management/client.md [hunk 2]
```diff
@@ -12,4 +12,4 @@ description: A container for client connection commands
 
 This is a container command for client connection commands.
 
-To see the list of available commands you can call `CLIENT HELP`.
\ No newline at end of file
+<!-- To see the list of available commands you can call `CLIENT HELP`. -->
```
Comment (dranikpg) [inline docs/command-reference/server-management/client.md:15]: We'd better add the help commands before 1.0

File: docs/command-reference/server-management/command.md [hunk 3]
```diff
@@ -69,65 +56,16 @@ Examples:
...
-
-Redis Cluster clients need to use other measures, as follows, to locate the keys for such commands.
-
-You can use the `COMMAND GETKEYS` command and have your Redis server report all keys of a given command's invocation.
-
-As of Redis 7.0, clients can use the [key specifications](#key-specifications) to identify the positions of key names.
-The only commands that require using `COMMAND GETKEYS` are `SORT` and `MIGRATE` for clients that parse keys' specifications.
-
-For more information, please refer to the [key specifications page][tr].
+// TODO: should we describe our internal flags?
```
Comment (dranikpg) [inline docs/command-reference/server-management/command.md:68]: should we describe things like reverse-mapping and no-autojournal? I guess they shouldn't be user-facing

File: docs/command-reference/server-management/hello.md [hunk 4]
```diff
@@ -6,24 +6,13 @@ description: Handshake with Redis
...
-when the connection is in this mode, Redis is able to reply with more semantical
-replies: for instance, `HGETALL` will return a *map type*, so a client library
-implementation no longer requires to know in advance to translate the array into
-a hash before returning it to the caller. For a full coverage of RESP3, please
-[check this repository](https://github.com/antirez/resp3).
-
-In Redis 6 connections start in RESP2 mode, so clients implementing RESP2 do
-not need to updated or changed. There are no short term plans to drop support for
-RESP2, although future version may default to RESP3.
+At the moment, the RESP3 protocol is not supported.
```
Comment (dranikpg) [inline docs/command-reference/server-management/hello.md:15]: HELLO currently makes little sense for switching protocols, maybe just remvoe the argument?

File: docs/command-reference/server-management/info.md [hunk 5]
```diff
@@ -11,240 +11,106 @@ description: Get information and statistics about the server
...
-used_cpu_sys:2574.705202
-used_cpu_user:10275.246584
-used_cpu_sys_children:0.001313
-used_cpu_user_children:0.002033
-used_cpu_sys_main_thread:2574.309989
-used_cpu_user_main_thread:10274.445298
-
-# Modules
-
-# Errorstats
```
Comment (dranikpg) [inline docs/command-reference/server-management/info.md:210]: In Df errors stats is not part of default

File: docs/command-reference/server-management/info.md [hunk 6]
```diff
@@ -261,13 +127,12 @@ INFO
...
 
-Please note depending on the version of Redis some of the fields have been
+Please note depending on the version of Dragonfly some of the fields have been
 added or removed. A robust client application should therefore parse the
 result of this command by skipping unknown properties, and gracefully handle
 missing fields.
 
-Here is the description of fields for Redis >= 2.4.
-
+<!-- TODO
```
Comment (dranikpg) [inline docs/command-reference/server-management/info.md:135]: 😢

File: docs/command-reference/server-management/role.md [hunk 7]
```diff
@@ -92,4 +92,5 @@ The sentinel output is composed of the following parts:
 ROLE
 ```
 
+// TODO: What about the slave note?
 **A note about the word slave used in this man page**: Starting with Redis 5, if not for backward compatibility, the Redis project no longer uses the word slave. Unfortunately in this command the word slave is part of the protocol, so we'll be able to remove such occurrences only when this API will be naturally deprecated.
```
Comment (dranikpg) [inline docs/command-reference/server-management/role.md:96]: Should we keep such a note that we use the term slave only for backward comp?

File: docs/command-reference/server-management/save.md [hunk 8]
```diff
@@ -6,20 +6,15 @@ description: Synchronously save the dataset to disk
...
+    SAVE [RDB|DF]
 
 **Time complexity:** O(N) where N is the total number of keys in all databases
 
-The `SAVE` commands performs a **synchronous** save of the dataset producing a
-_point in time_ snapshot of all the data inside the Redis instance, in the form
-of an RDB file.
+The `SAVE` commands performs a save of the dataset producing a
+_point in time_ snapshot of all the data inside the Dragonfly instance, in the form
+of a collection of DFS files.
```
Comment (dranikpg) [inline docs/command-reference/server-management/save.md:15]: collection of dfs files sounds not good

File: docs/command-reference/server-management/info.md [hunk 9]
```diff
@@ -610,52 +249,17 @@ The **commandstats** section provides statistics based on the command type,
...
 
-*   `dbXXX`: `keys=XXX,expires=XXX`
+*   `dbXXX`: `keys=XXX,expires=XXX, avg_ttl=XXX`
 
 [hcgcpgp]: http://code.google.com/p/google-perftools/
 
-**A note about the word slave used in this man page**: Starting with Redis 5, if not for backward compatibility, the Redis project no longer uses the word slave. Unfortunately in this command the word slave is part of the protocol, so we'll be able to remove such occurrences only when this API will be naturally deprecated.
-
-**Modules generated sections**: Starting with Redis 6, modules can inject their info into the `INFO` command, these are excluded by default even when the `all` argument is provided (it will include a list of loaded modules but not their generated info fields). To get these you must use either the `modules` argument or `everything`.,
+// TODO: slace question
```
Comment (dranikpg) [inline docs/command-reference/server-management/info.md:264]: .

File: docs/command-reference/server-management/info.md [hunk 10]
```diff
@@ -261,118 +127,45 @@ INFO
...
-     result of calling `UNLINK`, or `FLUSHDB` and `FLUSHALL` with the **ASYNC**
-     option)
-*   `lazyfreed_objects`: The number of objects that have been lazy freed.
+// TODO: Custom flags
 
+// TODO: How do make users check fragmentation ratio? `MEMORY MALLOC-STATS`?
+<!--
 Ideally, the `used_memory_rss` value should be only slightly higher than
 `used_memory`.
 When rss >> used, a large difference may mean there is (external) memory fragmentation, which can be evaluated by checking
```
Comment (dranikpg) [inline docs/command-reference/server-management/info.md:171]: we should add this to the docs

**Review by romange — COMMENTED** [review]
why did you delete "auth" command?

File: docs/command-reference/server-management/client-list.md (hunk 1 above)
Comment (romange) [reply docs/command-reference/server-management/client-list.md:34]: todo is fine. maybe `state of the connection as captured by "client list" command`

**Review by romange — COMMENTED** [review]
we also support "memory help"

File: docs/command-reference/server-management/save.md (hunk 8 above)
Comment (romange) [reply docs/command-reference/server-management/save.md:15]: "DFS" abbreviation must be explained. collection -> group -> set ?

File: docs/command-reference/server-management/role.md (hunk 7 above)
Comment (romange) [reply docs/command-reference/server-management/role.md:96]: We should remove this disclaimer.

File: docs/command-reference/server-management/role.md [hunk 11]
```diff
@@ -10,7 +10,7 @@ description: Return the role of the instance in the context of replication
 
 **Time complexity:** O(1)
 
-Provide information on the role of a Redis instance in the context of replication, by returning if the instance is currently a `master`, `slave`, or `sentinel`. The command also returns additional information about the state of the replication (if the role is master or slave) or the list of monitored master names (if the role is sentinel).
+Provide information on the role of a Dragonfly instance in the context of replication, by returning if the instance is currently a `master`, `slave`, or `sentinel`. The command also returns additional information about the state of the replication (if the role is master or slave) or the list of monitored master names (if the role is sentinel).
```
Comment (romange) [inline docs/command-reference/server-management/role.md:13]: we never return `sentinel`

**Review by romange — COMMENTED** [review]
Left a few commands.

File: docs/command-reference/server-management/save.md (hunk 8 above)
Comment (dranikpg) [reply docs/command-reference/server-management/save.md:15]: Lets call it 'set of DFS files' and make it point to the serialization page

File: docs/command-reference/server-management/command.md (hunk 3 above)
Comment (dranikpg) [reply docs/command-reference/server-management/command.md:68]: I left no description and instead filed an issue https://github.com/dragonflydb/dragonfly/issues/861

**Comment by dranikpg** [comment]
I added auth (by accident), cut out the ACL section
I added MEMORY HELP back but its not very helpful as we support one command and the other parts contains a "not supported" note

**Review by romange — APPROVED** [review]

**Pull request merged**